# [\#13794 Issue](https://github.com/ethereum/solidity/issues/13794) `open`: ICE when Yul sub-object name with a dot shadows the name of a nested object
**Labels**: `bug :bug:`, `low effort`, `low impact`, `must have eventually`


#### <img src="https://avatars.githubusercontent.com/u/137030?v=4" width="50">[cameel](https://github.com/cameel) opened issue at [2022-12-09 20:58](https://github.com/ethereum/solidity/issues/13794):

## Description
Sub-object names in Yul can contain dots. Dots are also used for accessing nested objects. If you try to do both at the same time so that the qualified name of a nested object matches another name that uses a dot, the compiler crashes with an ICE.

## Steps to Reproduce
### `test.yul`
```yul
object "A" {
    code {}

    object "B" {
        code {}

        object "C" {
            code {}
        }
    }

    object "B.C" {
        code {}
    }
}
```

### Output
```bash
solc --strict-assembly test.yul
```
```
Uncaught exception:
/solidity/libyul/Object.cpp(99): Throw in function std::set<solidity::yul::YulString> solidity::yul::Object::qualifiedDataNames() const
Dynamic exception type: boost::wrapexcept<solidity::yul::YulAssertion>
std::exception::what: Yul assertion failed
[solidity::util::tag_comment*] = Yul assertion failed
```

Note that there is no crash if you move the definition `B.C` above `B`.

## Environment
- Compiler version: 0.8.17




-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2024.12.15 at 06:45:24.]
